### PR TITLE
cms: fix coverity 1485981: unchecked return value

### PR DESCRIPTION
**Pull request by paulidale**

### CODE REVIEW

File: crypto/cms/cms_ec.c [hunk 1]
```diff
@@ -20,20 +20,23 @@ static EVP_PKEY *pkey_type2param(int ptype, const void *pval,
...
                                             selection, libctx, propq);
         if (ctx == NULL)
             goto err;
 
-        OSSL_DECODER_from_data(ctx, &pm, &pmlen);
+        if (!OSSL_DECODER_from_data(ctx, &pm, &pmlen)) {
+            ERR_raise(ERR_LIB_CMS, CMS_R_DECODE_ERROR);
+            goto err;
+        }    
         OSSL_DECODER_CTX_free(ctx);
```
Comment (slontis) [inline crypto/cms/cms_ec.c:40]: ctx = NULL; after the free?

File: crypto/cms/cms_ec.c [hunk 2]
```diff
@@ -60,6 +63,7 @@ static EVP_PKEY *pkey_type2param(int ptype, const void *pval,
  err:
     EVP_PKEY_free(pkey);
     EVP_PKEY_CTX_free(pctx);
```
Comment (slontis) [inline crypto/cms/cms_ec.c:65]: Why doesnt the pctx get freed above in it does return pkey?

File: crypto/cms/cms_ec.c (hunk 1 above)
Comment (paulidale) [reply crypto/cms/cms_ec.c:40]: Doesn't need it.

File: crypto/cms/cms_ec.c (hunk 2 above)
Comment (paulidale) [reply crypto/cms/cms_ec.c:65]: No idea.  @DDvO ?  @mattcaswell ?

Comment (t8m) [reply crypto/cms/cms_ec.c:65]: Simply a bug (a leak). I suppose we aren't testing that case.

Comment (DDvO) [reply crypto/cms/cms_ec.c:65]: I see this code for the first time, but the missing `EVP_PKEY_CTX_free(pctx);`
at the end of the first `else` block is clearly an oversight.

Comment (paulidale) [reply crypto/cms/cms_ec.c:66]: I've added the free call.

File: crypto/cms/cms_ec.c (hunk 1 above)
Comment (paulidale) [reply crypto/cms/cms_ec.c:40]: I've moved the return up to clarify things.

**Review by t8m — APPROVED** [review]
Approved if CI passes.

File: crypto/cms/cms_ec.c [hunk 3]
```diff
@@ -50,16 +54,17 @@ static EVP_PKEY *pkey_type2param(int ptype, const void *pval,
...
             goto err;
-    } else {
-        ERR_raise(ERR_LIB_CMS, CMS_R_DECODE_ERROR);
-        goto err;
+        EVP_PKEY_CTX_free(pctx);
+        return pkey;
     }
 
+    ERR_raise(ERR_LIB_CMS, CMS_R_DECODE_ERROR);
     return pkey;
```
Comment (DDvO) [inline crypto/cms/cms_ec.c:62]: A `return NULL` would be more clear here, but nevertheless okay.

**Review by DDvO — APPROVED** [review]

File: crypto/cms/cms_ec.c (hunk 3 above)
Comment (paulidale) [reply crypto/cms/cms_ec.c:62]: I'll change it.

**Review by DDvO — APPROVED** [review]
LGTM

**Comment by t8m** [comment]
Still OK.

**Comment by paulidale** [comment]
Merged to master, thanks for the reviews.

**Pull request closed without merging**